NameLayerMain/src/vg/civcraft/mc/namelayer/GroupManager.java

```diff
@@ -330,25 +330,21 @@ public boolean hasAccess(Group group, UUID player, PermissionType perm) {
 			}
 		}
 		GroupPermission perms = getPermissionforGroup(group);
-		for(PlayerType rank : getRecursivePlayerTypes(group, player)) {
-			if (perms.hasPermission(rank, perm)) {
-				//player has right rank in the group itself or at least one super group
-				return true;
-			}
-		}		
+		PlayerType rank = getRecursiveHighestPlayerType(group, player); 
+		if (perms.hasPermission(rank, perm)) {
+			//player has right rank in the group itself or at least one super group
+			return true;
+		}	
 		return false;
 	}
 	
-	private List<PlayerType> getRecursivePlayerTypes(Group group, UUID player) {
-		List<PlayerType> perms = new LinkedList<PlayerType>();
+	private PlayerType getRecursiveHighestPlayerType(Group group, UUID player) {
 		PlayerType type = group.getPlayerType(player);
-		if (type != null) {
-			perms.add(type);
-		}
 		if (group.hasSuperGroup()) {
-			perms.addAll(getRecursivePlayerTypes(group.getSuperGroup(), player));
+			//only return highest rank
+			type = PlayerType.getByID(Math.max(PlayerType.getID(type), PlayerType.getID(getRecursiveHighestPlayerType(group.getSuperGroup(), player))));
 		}
-		return perms;
+		return type;
```

Comment:
Why is the current way not good? It's more correct... and not really any less clear. "If your permission level on the subgroup, or the permission level from any supergroup, grants you permission, you have permission. Otherwise, you don't".

The only other thing I guess you could do if you really don't like the old code and want to enforce hierarchy is to backfill "lower" types when doing the check -- e.g. if you are MODS, check against MODS and MEMBERS, etc.

I'd recommend sticking with the prior solution, however, which has the advantage of already handling every edge case and not enforcing hierarchies unintentionally or otherwise.